lib/datasets/libsvm-dataset-list.rb

```diff
@@ -78,6 +78,13 @@ def extract_text(element)
     def open_detail(detail)
       data_path = cache_dir_path + detail
       download(data_path, @metadata.url + detail)
+      # Hack to fix XML error in EUR-Lex section of
+      # https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/multilabel.html
+      if detail == "multilabel.html"
+        page = data_path.read
+        page.gsub!("'>', '<',", "'&gt', '&lt',")
+        data_path.write(page)
```

Comment:
The downloaded file is cached.
So we should not write substituted content.

Could you substitute the content in memory?

```ruby
data_path.open do |input|
  if detail == "multilabel.html"
    yield(input.gsub("'>', '<',", "'&gt', '&lt',"))
  else
    yield(input)
  end
end
```

Reply:
The site has been updated. Maybe this change can be reverted? Can also look at later updating the code to parse HTML.

Reply:
While cleaner input.gsub fails with 
```
Error: test: multi-label(LIBSVMDatasetTest): NoMethodError: undefined method `gsub' for #<File:0x00007f90436c0bf8>
```
input is a file object, so may need some other way to modify it. As the site has been updated, can revert to original code.

Reply:
Ah, sorry. `.read` was missed: `input.read.gsub(...)`